dali/operators/generic/slice/slice_attr.h

```diff
@@ -199,6 +184,86 @@ class SliceAttr {
       };
   }
 
+ private:
+  const OpSpec &spec_;
+
+  std::vector<int> axes_;
+  TensorLayout axis_names_;
+
+  ArgValue<int, 1> start_;
+  ArgValue<float, 1> rel_start_;
+
+  ArgValue<int, 1> end_;
+  ArgValue<float, 1> rel_end_;
+
+  ArgValue<int, 1> shape_;
+  ArgValue<float, 1> rel_shape_;
+
+  std::vector<CropWindowGenerator> crop_window_generators_;
+
+  bool has_start_, has_end_, has_shape_;
+};
+
+
+class PositionalSliceAttr {
+ public:
+  explicit inline PositionalSliceAttr(const OpSpec &spec)
+      : spec_(spec),
+        normalized_anchor_(spec.GetArgument<bool>("normalized_anchor")),
+        normalized_shape_(spec.GetArgument<bool>("normalized_shape")),
```

Comment:
Nitpick: I think that initialization of non-const fields of fundamental type is more readable when it's done by simple assignment in the constructor body.